code/modules/cargo/packs/organic.dm

```diff
@@ -185,52 +181,58 @@
 		/obj/item/food/pizza/pineapple = 10,
 		/obj/item/food/pizza/arnold = 3,
 		/obj/item/food/pizza/energy = 5
-	) //weighted by chance to disrupt eaters' rounds
+	)
 
-	for(var/obj/item/pizzabox/P in C)
-		if(!anomalous_box_provided)
-			if(prob(anna_molly_box_chance)) //1% chance for each box, so 4% total chance per order
-				var/obj/item/pizzabox/infinite/fourfiveeight = new(C)
-				fourfiveeight.boxtag = P.boxtag
-				fourfiveeight.boxtag_set = TRUE
-				fourfiveeight.update_appearance()
-				qdel(P)
-				anomalous_box_provided = TRUE
-				log_game("An anomalous pizza box was provided in a pizza crate at during cargo delivery.")
-				if(prob(50))
-					addtimer(CALLBACK(src, .proc/anomalous_pizza_report), rand(300, 1800))
-					message_admins("An anomalous pizza box was provided in a pizza crate at during cargo delivery.")
-				else
-					message_admins("An anomalous pizza box was silently created with no command report in a pizza crate delivery.")
-				continue
+/datum/supply_pack/organic/pizza/fill(obj/structure/closet/crate/new_crate)
+	. = ..()
+	var/list/rng_pizza_list = pizza_types.Copy()
+	for(var/i in 1 to 5)
+		if(add_anomalous(new_crate))
+			continue
+		if(add_boombox(new_crate))
+			continue
+		add_normal_pizza(new_crate, rng_pizza_list)
 
-		if(!boombox_provided)
-			var/boombox_lottery = rand(1,boombox_tickets)
-			var/boombox_type
-			switch(boombox_lottery)
-				if(1 to 2)
-					boombox_type = /obj/item/pizzabox/bomb/armed //explodes after opening
-				if(3)
-					boombox_type = /obj/item/pizzabox/bomb //free bomb
+/// adds the chance for an infinite pizza box
+/datum/supply_pack/organic/pizza/proc/add_anomalous(obj/structure/closet/crate/new_crate)
+	if(anomalous_box_provided)
+		return FALSE
+	if(!prob(infinite_pizza_chance))
+		return FALSE
+	new /obj/item/pizzabox/infinite(new_crate)
+	anomalous_box_provided = TRUE
+	log_game("An anomalous pizza box was provided in a pizza crate at during cargo delivery.")
+	if(prob(50))
+		addtimer(CALLBACK(src, .proc/anomalous_pizza_report), rand(300, 1800))
```

Comment:
```suggestion
		addtimer(CALLBACK(src, .proc/anomalous_pizza_report), rand(30 SECONDS, 180 SECONDS))
```
labelling units helps people know the exact intent, may as well touch it up while you're doing stuff ehre